### DISCUSSION TITLE
How to implement on NFS

### DISCUSSION THREAD

**Discussion by fdfytr** [Luctia/ezarr#23] [Help wanted]
Would appreciate instructions/step on how to run it on NFS volume

**Comment by fdfytr** [comment]
I had managed to install package on nfs mount. Containers are running but nothing in web-browser. I had checked ports and docker logs... What can the problem be?

**Comment by Luctia** [comment]
What address are you trying to use?

**Comment by fdfytr** [comment]
> What address are you trying to use?

the one from ip addr

**Comment by Luctia** [comment]
So xxx.xxx.xxx.xxx:8686 for example? Do you have the relevant ports open on your server?

**Comment by fdfytr** [comment]
> So xxx.xxx.xxx.xxx:8686 for example? Do you have the relevant ports open on your server?

yep, ie for qbittorrent
`ss -tulpn | grep LISTEN | grep :8080
tcp   LISTEN 0      4096         0.0.0.0:8080       0.0.0.0:*
tcp   LISTEN 0      4096            [::]:8080          [::]:*`

it works on local volume, but empty webpage once i put data+docker on nfs (edit: i mean during compose up)...

**Comment by fdfytr** [comment]
I'd waited little bit longer and jackett popped up
but sonarr/radarr are not
jellyfin is throwing 
`jellyfin     | Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 5: 'database is locked`
looks as typical docker and nfs still [cannot get together](https://github.com/jellyfin/jellyfin/issues/3667) :(

**Comment by fdfytr** [comment]
I managed to put configs on local volume and data on nfs with adjusting permissions accordingly
found [this explanation](https://wiki.servarr.com/sonarr/faq#i-am-getting-an-error-database-disk-image-is-malformed) for sonarr

**Comment by fdfytr** [comment]
i've tried jellyfin from a different machine to get to /data/media subfolders on nfs
puid=0 pgid=13000 umask=002 but subfolders are not accessible
any tips on troubleshooting?

**Comment by dmgolembiowski** [comment]
@fdfytr I'm attempting this now; if I'm able to mitigate the issues you're experiencing I'll chime in. Assuming all things normally work fine with your NFS mount point (outside of any docker containers), I'm naively thinking it'll be necessary to use either a volume or a [bind mount](https://docs.docker.com/storage/bind-mounts/) to the file system. 

On my system, I'm running an NFS mount resembling:

```sh
# /etc/fstab
{{ NAS HOST }}:{{ NAS REMOTE PATH }} {{ LOCAL PATH }} nfs rw,suid,dev,no_root_squash,no_subtree_check,relatime,vers=3,rsize=524288,wsize=524288,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,mountaddr={{ NAS HOST }},mountvers=3,mountproto=udp,local_lock=none,addr={{ NAS HOST }} 0 0
``` 

I believe this working config gives all users write access, which seems like it ought to include docker and any transient users in the containers. All told, I haven't dug into this yet, but there's a chance you might be able to tweak your mount options to make things just work.

**Comment by Luctia** [comment]
@dmgolembiowski I reckon creating a volume to do this should be possible. I know it's pretty simple for smb, I don't have experience for NFS but it's probably possible. Good luck!

**Comment by dmgolembiowski** [comment]
Confirmed! (Sorry) 

It can be done and it does work. These are the pitfalls you have to watch out for:

- firewall rules
- inconsistent mount client and server config files

Honestly though, SSHFS was more fault tolerant and caused fewer file fragmentations under heavy + consistent traffic.

Who wants to build a fs with me?
